### ISSUE TITLE
Dynamically load UI implementations for Kubernetes CRDs

### ISSUE THREAD

**Issue by mclarke47** [backstage/backstage#5511]
In https://github.com/backstage/backstage/pull/4892 we added the ability to have Kubernetes Custom Resource Definitions (CRDs) show up in the Kubernetes plugin.

However this implementation only really works for open source CRDs. If a user has their own closed source CRD it doesn't make sense to add the UI implementation to the Kubernetes plugin as it cannot be reused by other users.

So we need some mechanism for adding CRD UI implementations not present inside the plugin.

Perhaps this could be done by Backstage "scanning" for CRD React components present in provided packages? That is just one suggestion but I would like to hear alternatives.

**Comment by timbonicus** [comment]
Maybe there's some similarity to the proposed approach for [adding custom fields](https://github.com/backstage/backstage/pull/5443#discussion_r621634543) for the ScaffolderPage?

**Comment by freben** [comment]
Yeah, using the extensions mechanism for this as mentioned by @timbonicus sounds like a good approach. Do reach out to us if you start exploring it since it's still fairly early days for that system.

**Comment by mclarke47** [comment]
Still relevant
